### PR TITLE
try to infer the git rev from metadata when running in AWS ECS

### PR DESCRIPTION
**Pull request by jfeltesse-mdsol**
All in the title.

@JordiPolo

### CODE REVIEW

**Comment by cabbott** [comment]
lgtm

File: lib/kapnismology/application_information.rb [hunk 1]
```diff
@@ -11,12 +15,26 @@ def trace_id
...
+      result = `#{GIT_SHOW_COMMAND}`
+      $?.success? ? result : nil
+    end
+
+    # This assumes the ImageName features the git ref as the last part of a string delimited by dots.
+    # See spec/support/ecs_metadata.json as well as
+    # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html
+    def latest_ref_from_ecs_metadata
+      return unless !ECS_CONTAINER_METADATA_FILE&.strip.empty? && File.readable?(ECS_CONTAINER_METADATA_FILE)
+      JSON.parse(File.read(ECS_CONTAINER_METADATA_FILE))["ImageName"].split(".").last[0...7]
```
Comment (JordiPolo) [inline lib/kapnismology/application_information.rb]: low chances this will change soon but could we catch json parsing exceptions and return nil?

Comment (jfeltesse-mdsol) [reply lib/kapnismology/application_information.rb]: https://github.com/JordiPolo/kapnismology/pull/14/commits/daf149a684ff0f543b66baea17cb2e153130ee73

File: lib/kapnismology/application_information.rb [hunk 2]
```diff
@@ -1,22 +1,47 @@
...
+
+    # This assumes the ImageName features the git ref as the last part of a string delimited by dots.
+    # See spec/support/ecs_metadata.json as well as
+    # https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html
+    def latest_ref_from_ecs_metadata
+      return if ECS_CONTAINER_METADATA_FILE.nil?
+      return unless !ECS_CONTAINER_METADATA_FILE.strip.empty? && File.readable?(ECS_CONTAINER_METADATA_FILE)
+
+      begin
+        JSON.parse(File.read(ECS_CONTAINER_METADATA_FILE))["ImageName"].split(".").last[0...7]
```
Comment (JordiPolo) [inline lib/kapnismology/application_information.rb]: sorry, that's what happens when I comment late at night. I also meant to say do `&.` or in other way make this not blow up if the format is not exactly as expected. Just return nil. Maybe even log if the information can't be found

Comment (jfeltesse-mdsol) [reply lib/kapnismology/application_information.rb]: This kind of errors are caught in the calling method that has a "catch all" rescue.
Thinking of which I guess it doesn't make much sense to do that begin/rescue here if we're not going to do anything with that specific error.

**Pull request merged**